streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableTransformValues.java

```diff
@@ -134,13 +135,17 @@ public void close() {
 
         @Override
         public void init(final ProcessorContext context) {
+            this.context = context;
             parentGetter.init(context);
             valueTransformer.init(new ForwardingDisabledProcessorContext(context));
         }
 
         @Override
-        public V1 get(final K key) {
-            return valueTransformer.transform(key, parentGetter.get(key));
+        public ValueAndTimestamp<V1> get(final K key) {
+            final ValueAndTimestamp<V> valueAndTimestamp = parentGetter.get(key);
+            return ValueAndTimestamp.make(
+                valueTransformer.transform(key, getValueOrNull(valueAndTimestamp)),
+                valueAndTimestamp == null ? context.timestamp() : valueAndTimestamp.timestamp());
```

Comment:
I don't think so. Note, this is a lookup into a store of a different upstream processor (in case, the KTable is not materialized). Thus, we should only return the timestamp that is in the store. It's not an output record we handle here, but an input record that we look up.

Reply:
Agreed.

On the opposite side of the boolean, if the `valueAndTimestamp` _is_ null, should we just return `null`?

Reply:
> Agreed.

With whom?

> On the opposite side of the boolean, if the valueAndTimestamp is null, should we just return null?

 Why? If it's `null`, we should still pipe it through `transform` IMHO, and return whatever `transform()` returns.

Reply:
> Agreed.

Sorry, with you :)

On the `null` question: I see. There are some sketchy semantics around this, but that's not relevant to timestamps, but table/transform semantics in general, so we don't need to worry about it here.

I actually failed to remember that `make` returns `null` if the value is `null`, so this code already does what I was really concerned about. I thought it would return a `ValueAndTimestamp(null, context.timestamp())` if it got a null input, which seemed wrong. Now I see it will just return `null`, which I think is right.

So, we actually only return `context.timestamp()` if the transformer returns a non-null result for a null input. It seems a little strange to me, since `context.timestamp()` is actually unrelated to `key`'s time domain, but it probably doesn't matter, since:
* even with this change, we still are not actually _defining_ and time behavior, so we can do literally anything we want
* this method is currently only used for joins, where the resulting time would go on to be maxed with `context.timestamp()` anyway, resulting in `max(context.timestamp(), context.timestamp())`, which is trivially correct.

Reply:
Ok. Updating this to `UNKNOWN`.